src/App.tsx

```diff
@@ -1,95 +1,198 @@
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useState, useCallback, useMemo, useEffect } from 'react';
+import cn from 'classnames';
+import './styles/App.scss';
+import {
+  ContextPropsFilteredTodos,
+  ContextPropsMyTodos,
+  Tabs,
+  Todo,
+  handledTabs,
+  useLocalStorage,
+} from './utils/helpers';
+import { TodoList } from './components/TodoList/TodoList';
+import { FilteredTodos, MyTodos } from './utils/GlobalContext';
+
+export function App() {
+  const [todos, setTodos] = useLocalStorage<Todo[]>('todos', []);
+  const [filteredTodos, setFilteredTodos] = useState(todos);
+  const [input, setInput] = useState('');
+  const [formKey, setFormKey] = useState(0);
+  const [currentTab, setCurrentTab] = useState(Tabs.All);
+
+  const todosLeft = todos.filter((t: Todo) => !t.completed).length;
+  const areThereCompletedTodos = todos.some((t: Todo) => t.completed);
+
+  useEffect(() => {
+    switch (currentTab) {
+      case Tabs.Active:
+        setFilteredTodos(todos.filter((t: Todo) => !t.completed));
+        break;
+      case Tabs.Completed:
+        setFilteredTodos(todos.filter((t: Todo) => t.completed));
+        break;
+      default:
+        setFilteredTodos(todos);
+    }
+  }, [todos, currentTab]);
+
+  const handleInput = useCallback((e: React.ChangeEvent<HTMLInputElement>) => {
+    setInput(e.target.value);
+  }, []);
+
+  const handleFormSubmission = (e: React.ChangeEvent<HTMLFormElement>) => {
+    e.preventDefault();
+
+    if (input.trim() && input.length < 42) {
+      setTodos([
+        ...todos,
+        {
+          id: new Date().getTime(),
+          title: input,
+          completed: false,
+        },
+      ]);
+      setInput('');
+      setFormKey(n => n + 1);
+    } else {
+      if (input.length >= 43) {
+        // eslint-disable-next-line no-alert
+        alert('The max allowed number of characters is 43');
+      } else {
+        // eslint-disable-next-line no-alert
+        alert('Title should not be empty');
+      }
+
+      setInput('');
+      setFormKey(n => n + 1);
+    }
+  };
+
+  const myTodos = useMemo(
```

Comment:
If you want to add context create an additional component only with context logic, which returns something like this:
```
return (
  <ProviderExample value={value}>
    <TodoApp />
  </ProviderExample>
);
```